sql/moz-fx-data-shared-prod/firefox_ios_derived/first_28_day_retention_v1/query.sql

```diff
@@ -0,0 +1,37 @@
+WITH first_28_day_activity AS (
+  SELECT
+    client_id,
+    sample_id,
+    country,
+    mozfun.bits28.retention(days_seen_bits, submission_date) AS retention_first_28_days,
+    days_seen_bits,
+  FROM
+    firefox_ios_derived.clients_last_seen_joined_v1
+  WHERE
+    -- need to make submission_date date 1 day older due to 2 day lag inside the table
+    submission_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+),
+corresponding_first_seen_clients AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+  FROM
+    firefox_ios.baseline_clients_first_seen
+  WHERE
+    -- 29 instead of 28 to compensate for the 2 day lag inside clients_last_seen_joined_v1
+    submission_date = DATE_SUB(@submission_date, INTERVAL 29 DAY)
+)
+SELECT
+  client_id,
+  sample_id,
```

Comment:
We've seen dupes in android tables, can we guarantee that there won't be any here? (for e.g. `firefox_ios_clients`, we join with the previous data. Maybe here we'd opt instead for just a `check.sql` to verify we're not getting dupes?)

Reply:
Yup, I think that's reasonable here